gobluestingray/reporting-engine
# Feature/15.0/batch report printing

[#958696098 Batch report printing](https://www.wrike.com/workspace.htm?acc=5659506#todo?filters=keyword%3Dmywork%26status%3Dactive&sidePanelItemId=958696098&sortOrder=11)

Depends on PR #2 

Changes:
 - Adds a new Server Action, Batch Print Reports, for Sale Order and Product Template models
 - Adds a new wizard, Batch Print Report Wizard, that is launched by the Server Action
   - All HTML reports for the model are available for batching
   - Adds the option to collate reports by record instead of report
   
To test:
 - Clone `gobluestingray/reporting-engine`
 - Add the `report_batch` module to to the 15.0 branch of `blue_odoo_addons`'s docker-compose.base.yml (in services.volumes)
   - i.e. `../<reporting-engine repo name>/report_batch:/opt/odoo/vendor/report_batch`
 - Add [sample sale reports](https://github.com/gobluestingray/metricwise_odoo/pull/5/commits/811e7df90914857e2bedeee698cfb203c031602c#diff-69f6af1c82761a23d5881ae84ece84e0e05211d1234c0434c89510502f8cce3f) and [templates](https://github.com/gobluestingray/metricwise_odoo/pull/5/commits/811e7df90914857e2bedeee698cfb203c031602c#diff-74b7a7e0356fe940ba9ca90f10b9318987a823a9aa6a66c66991189b900d6ae7) to the `sale` module in the 15.0 branch of `blue_odoo_addons`
 - Start a new db from the 15.0 branch of `blue_odoo_addons`, install the `report_batch` (and the "Sales") module(s), and check out the new functionality
 
Remaining Issues:
 - Report header still "prints" 
 - "Instructions..." link in report header needs working link
 - Report opens in same tab
 - "Collate by record" only works on Sale Orders (not Products): need better view munging to get document name consistently across models
 - Printing the generated batch report results in a blank report (only the Report header prints)
   - I investigated a few things that I thought may be causing this, with no success:
     - The report header isn't causing this: I removed the header completely and the report was still blank
     - I tried adding `active_ids` to the context, but the report was still blank
     - I don't _think_ it's caused by my sample reports being mostly blank
